### PR TITLE
EL-3376 - Dynamic Notification Height

### PR DESCRIPTION
**Pull request by ashley-hunter**
Related PR: https://github.houston.softwaregrp.net/caf/ux-aspects-micro-focus/pull/337

**Notifications**
- Notifications are now sized dynamically, there is now no need to specify the height. Note the height property is there to specify a height is desired.
- In the ticket it talked about the possibility of enforcing a max height, although I found this to not really be necessary due to the notification content being a completely custom template, it makes more sense for the classes in the template to apply any height limitations, also setting the existing height property would restrict the height.
- Additionally there was an issue when notifications are displayed on the bottom of the page when they dont move up they just get push off the page:
https://i.gyazo.com/df3292adde9a9b849342741656ba2968.mp4
This has now been fixed.

**Color Service**
I also made a few improvements to the color service:
- Split into multiple files as color sets, interfaces and multiple classes were all crammed into one file
- We were doing some stuff creating elements and get computed styles etc, if an application was using server side rendering or running unit tests in the console this would break, so I have changed this to use color values the same as we are doing for the micro focus colorSet. This also allowed the color service private functions to be greatly simplified and be faster.
- I have also added a fix for the lazy module issue. Currently the application must set the colorSet in every lazy loaded module. This is not an issue any more as it checks for a parent instance and if found it reuses it. This also allowed the addition of a forRoot function so this can simply be set once in the app module:
```typescript
ColorServiceModule.forRoot(colorSets.microFocus)
```

#### Ticket
https://autjira.microfocus.com/browse/EL-3376

#### Documentation CI URL
https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_EL-3376-Dynamic-Notification-Height

### CODE REVIEW

**Comment by donna-marijne** [comment]
This doesn't seem to have built correctly, can you check?

https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_EL-3376-Dynamic-Notification-Height/#/components/notifications#notifications

**Comment by ashley-hunter** [comment]
Thanks, that should be it working now.

File: docs/app/pages/components/components-sections/notifications/notifications/snippets/app.html [hunk 1]
```diff
@@ -38,43 +38,43 @@ <h4 class="notification-title">Messages Received</h4>
...
+        <ux-accordion>
+            <ux-accordion-panel class="accordion-chevron" heading="Customize Example...">
+                <div class="row uxd-customize-row">
+                    <div class="col-md-6 col-sm-12">
+                        <label for="backgroundColor">backgroundColor</label>
+                        <div class="input-group m-b-sm">
+                            <input type="text" placeholder="Enter color" class="form-control"
+                                [(ngModel)]="backgroundColor">
+                            <span class="input-group-addon color-display"
+                                [style.background-color]="backgroundColor"></span>
```
[the remark this answers is not in the document]
Comment (ashley-hunter) [reply docs/app/pages/components/components-sections/notifications/notifications/snippets/app.html]: Sure, no problem at all! I will make that change, thanks!

**Comment by ashley-hunter** [comment]
Review fixes made

**Pull request merged**